### PR TITLE
feat: add logger service

### PR DESCRIPTION
**Pull request by Hitesh-Kandpal**
Changes in PR:
1. Add logger service

## Summary by CodeRabbit

- **New Features**
	- Introduced a centralized logging service that enhances the application's logging capabilities with structured and flexible logging options.
	- Added enumerations for log levels and severity, standardizing logging practices across the application.
	- Added various logging methods for different severity levels: info, warn, error, and fatal, enabling improved monitoring and debugging.
	- Each log entry is formatted consistently, making it easier to track and analyze logs associated with user sessions.

### CODE REVIEW

File: src/app/core/services/logger.service.ts [hunk 1]
```diff
@@ -0,0 +1,159 @@
+import { Injectable } from '@angular/core';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class LoggerService {
+  private sessionId = 'sample-session-id'; // Replace with actual session ID logic
```
Comment (xixas) [inline src/app/core/services/logger.service.ts:7]: Add a method here to  generate a session Id too.

**Review by xixas — APPROVED** [review]

File: src/app/core/services/logger.service.ts [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+    tracebackId: string,
+    screenName: string,
+    source: string,
+    deviceInfo: string,
+    data: string,
+    message: string,
+    stackTrace?: string,
+  ): void {
+    this.log(
+      'WARN',
```
Comment (xixas) [inline src/app/core/services/logger.service.ts:102]: All the values like log level, severity should come from enum values.

**Review by xixas — CHANGES_REQUESTED** [review]

File: src/app/core/services/logger.service.ts [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+      message,
+      stackTrace,
+    });
+  }
+
+  log(
+    level: string,
+    severity: string,
+    tracebackId: string,
+    screenName: string,
```
Comment (xixas) [inline src/app/core/services/logger.service.ts:42]: THis should handle filling up most generic fields like screenName, deviceInfo, source, timestamp etc.
When in use we only add message and stacktrace. Also what is the plan/idea for tracebackId?

**Review by xixas — CHANGES_REQUESTED** [review]

File: src/app/core/services/logger.service.ts (hunk 1 above)
Comment (HiteshGehu) [reply src/app/core/services/logger.service.ts]: added a method for generating session id

File: src/app/core/services/logger.service.ts (hunk 2 above)
Comment (HiteshGehu) [reply src/app/core/services/logger.service.ts]: added enum values for log level and severity and used them in logger service

File: src/app/core/services/logger.service.ts (hunk 3 above)
Comment (HiteshGehu) [reply src/app/core/services/logger.service.ts]: wrote function for filling these generic fields

The tracebackId will be a unique identifier for each log entry, helping us trace and correlate related logs. It's generated using a combination of the current timestamp and a random value to ensure uniqueness.
(I saw this was added in initial setup for pickleball logger service file, so I added it here too)

**Review by xixas — APPROVED** [review]

**Pull request merged**